extensions/cloud/portability-cloud-microsoft/src/main/java/org/dataportabilityproject/cloud/microsoft/cosmos/AzureTableStore.java

```diff
@@ -0,0 +1,285 @@
+  private static final String ENDPOINT_TEMPLATE = "https://%s.table.cosmosdb.azure.com:443/";
+  private static final String BLOB_CONNECTION_TEMPLATE =
+      "DefaultEndpointsProtocol=https;AccountName=%s;AccountKey=%s;";
+
+  private static final String JOB_TABLE = "DP_JOBS";
+  private static final String JOB_DATA_TABLE = "DP_JOB_DATA";
+
+  private static final String BLOB_CONTAINER =
+      "dataportability"; // Azure rules: The container name must be lowercase
+  private static final int UNKNOWN_LENGTH = -1;
+
+  private final TableStoreConfiguration configuration;
+
+  private CloudTableClient tableClient;
+  private CloudBlobClient blobClient;
+
+  public AzureTableStore(TableStoreConfiguration configuration) {
+    this.configuration = configuration;
+  }
+
+  public void init() {
+    try {
+      String endpoint = String.format(ENDPOINT_TEMPLATE, configuration.getAccountName());
+
+      CloudStorageAccount cosmosAccount =
+          CloudStorageAccount.parse(
+              String.format(
+                  COSMOS_CONNECTION_TEMPLATE,
+                  configuration.getAccountName(),
+                  configuration.getAccountKey(),
+                  endpoint));
+      tableClient = cosmosAccount.createCloudTableClient();
+
+      // Create the tables if the do not exist
+      tableClient.getTableReference(JOB_TABLE).createIfNotExists();
+      tableClient.getTableReference(JOB_DATA_TABLE).createIfNotExists();
+
+      CloudStorageAccount blobAccount =
+          CloudStorageAccount.parse(
+              String.format(
+                  BLOB_CONNECTION_TEMPLATE,
+                  configuration.getAccountName(),
+                  configuration.getBlobKey()));
+      blobClient = blobAccount.createCloudBlobClient();
+
+      blobClient.getContainerReference(BLOB_CONTAINER).createIfNotExists();
+    } catch (StorageException | URISyntaxException | InvalidKeyException e) {
+      throw new MicrosoftStorageException(e);
+    }
+  }
+
+  @Override
+  public void createJob(UUID jobId, PortabilityJob job) throws IOException {
+    Preconditions.checkNotNull(jobId, "Job id is null");
+    Preconditions.checkNotNull(job, "Job is null");
+
+    try {
+
+      create(jobId, JOB_TABLE, job.jobAuthorization().state().name(), job);
+
+    } catch (JsonProcessingException e) {
+      throw new IOException(e);
+    }
+  }
+
+  @Override
+  public void updateJob(UUID jobId, PortabilityJob job) throws IOException {
+    updateJob(jobId, job, null);
+  }
+
+  @Override
+  public void updateJob(UUID jobId, PortabilityJob job, JobUpdateValidator validator)
+      throws IOException {
+    Preconditions.checkNotNull(jobId, "Job not persisted");
+    Preconditions.checkNotNull(job, "Job is null");
+    try {
+
+      CloudTable table = tableClient.getTableReference(JOB_TABLE);
+
+      String serializedJob = configuration.getMapper().writeValueAsString(job);
+      DataWrapper wrapper =
```

Comment:
passing the jobID as the "rowKey" here makes jobID the key, right? Would be useful to add a comment about this